### PR TITLE
Make pup NOLINTs consistent

### PR DESCRIPTION
**Pull request by nilsvu**
This will hopefully reduce some noise in PR reviews.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by wthrowe — DISMISSED** [review]
Needs a rebase.

**Comment by nilsvu** [comment]
@wthrowe done

**Review by wthrowe — DISMISSED** [review]

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DhGaugeParameters.hpp [hunk 1]
```diff
@@ -134,7 +134,8 @@ struct DhGaugeParameters<false> {
   DhGaugeParameters() = default;
 
   // clang-tidy: google-runtime-references
```
Comment (nikwit) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DhGaugeParameters.hpp]: delete line

File: src/ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp [hunk 2]
```diff
@@ -95,9 +93,8 @@ class And : public Trigger {
   }
 
   // clang-tidy: google-runtime-references
```
Comment (nikwit) [inline src/ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp]: delete

File: src/ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp [hunk 3]
```diff
@@ -133,9 +130,8 @@ class Or : public Trigger {
...
```
Comment (nikwit) [inline src/ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp]: delete

File: src/Utilities/TaggedTuple.hpp [hunk 4]
```diff
@@ -320,7 +318,8 @@ class TaggedTuple : private tuples_detail::TaggedTupleLeaf<Tags>... {  // NOLINT
   static constexpr size_t size() { return sizeof...(Tags); }
 
   // clang-tidy: runtime-references
```
Comment (nikwit) [inline src/Utilities/TaggedTuple.hpp:320]: delete

**Review by nikwit — CHANGES_REQUESTED** [review]
some duplicates

**Comment by nilsvu** [comment]
Fixed, thanks @nikwit 👍

**Review by nikwit — COMMENTED** [review]
Looks good

**Review by nikwit — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]
Clang-tidy failures are Boost.

**Pull request merged**